packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelLoweringExtension.kt

```diff
@@ -57,6 +57,14 @@ private class RealmModelLowering(private val pluginContext: IrPluginContext) : C
             if (irClass.isData) {
                 error("Data class '${irClass.kotlinFqName}' is not currently supported.")
             }
+            // We don't support object
+            if (irClass.isAnonymousObject) {
+                error("Object '${irClass.parent.kotlinFqName}' is not supported. Only normal classes can inherit from 'RealmObject' or 'EmbeddedRealmObject'.")
```

Comment:
I am calling them just objects but could also call it anonymous objects, I feel like the former is clearer considering the Kotlin keyword. also, because of the `irClass` seems to work with objects, I am referring to its parent's name here, which does yield the object variable name. I also assume we won't be supporting this in the future?

Reply:
It looks like you added support for detecting anonymous objects instead 😄 (which is great, we should probably have that anyway), like here: https://kotlinlang.org/docs/object-declarations.html#creating-anonymous-objects-from-scratch

So I would rename this to `Anonymous objects classes cannot inherit from RealmObject. Only normal ...`. I would not link to the parent name, that is just confusing IMO, and the compiler should have a link to the line as part of the error.

I was actually thinking about `object declarations` like here: https://kotlinlang.org/docs/object-declarations.html#object-declarations-overview

Not 100% sure how to phrase that well though, maybe something like `Object declarations cannot inherit from 'RealmObject' or 'EmbeddedRealmObject' cannot ....` 

That means the name of the object class will not be part of the name, but maybe that is acceptable.